cg/services/sample_service/sample_service.py

```diff
@@ -0,0 +1,43 @@
+from cg.services.sample_service.utils import (
+    get_cancel_comment,
+    get_confirmation_message,
+)
+from cg.store.models import User
+from cg.store.store import Store
+
+
+class SampleService:
+
+    def __init__(self, store: Store):
+        self.store = store
+
+    def cancel_sample(self, sample_id: int, user_email: str) -> list[str]:
+        """Cancel a single sample and return a list of all cases it used to be in."""
+        case_ids: list[str] = self.store.get_case_ids_with_sample(sample_id)
+
+        self.store.mark_sample_as_cancelled(sample_id)
+        self.store.delete_sample_from_cases(sample_id)
+        self.store.delete_cases_without_samples(case_ids)
```

Comment:
I don't think these two steps are reversible, so I do not think that this meets the acceptance criteria listed by @beatrizsavinhas:

![image](https://github.com/user-attachments/assets/8a9fc4e7-4f87-4d3f-b4cc-4a006eb033dc)

Reply:
This PR has nothing to do with those criteria, it just refactors the existing cancel action.